src/app/[affiliateSlug]/cart/CartPage.tsx

```diff
@@ -6,18 +6,37 @@ import { CartItem, Carts } from "@/types/cart";
 import { faTrash } from "@awesome.me/kit-ac6c036e20/icons/classic/regular";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import { debounce } from "lodash";
-import { useState } from "react";
+import { useEffect, useState } from "react";
+import { useCookies } from "react-cookie";
 
 interface CartPageProps {
   cartInfo: Carts;
   slug: string;
 }
 
 export default function CartPage({ cartInfo, slug }: CartPageProps) {
-  const [cart, setCart] = useState<CartItem[]>(cartInfo?.cart_items ?? []);
+  const [cart, setCart] = useState<CartItem[] | []>(cartInfo?.cart_items ?? []);
   const [subtotal, setSubtotal] = useState<string>(
     cartInfo?.sub_total ?? "0.00"
   );
+  const [cookie, setCookie] = useCookies();
+
+  const refetchCart = async () => {
+    try {
+      const response = await fetch("/api/cart");
+      if (!response.ok) throw new Error("Failed to fetch cart");
+
+      const updatedCart = await response.json();
+      setCart(updatedCart.cart_items);
+      setSubtotal(updatedCart.sub_total);
+    } catch (error) {
+      console.error("Error fetching cart:", error);
+    }
+  };
+
+  useEffect(() => {
+    refetchCart();
```

Comment:
this is to sync the state if the widget adds more to the cart (by pressing +)?

Reply:
Yes, its for quantity changes sync